cpp/src/parquet/metadata.cc

```diff
@@ -690,8 +690,9 @@ class FileMetaData::FileMetaDataImpl {
   }
 
   void AppendRowGroups(const std::unique_ptr<FileMetaDataImpl>& other) {
-    if (!schema()->Equals(*other->schema())) {
-      throw ParquetException("AppendRowGroups requires equal schemas.");
```

Comment:
I would keep this original sentence as the start of the new message (it's still useful to know the context of why the schema difference is printed)